src/Umbraco.Core/Sync/LastSyncedManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Umbraco.Cms.Core.Persistence.Repositories;
+using Umbraco.Cms.Core.Scoping;
+
+namespace Umbraco.Cms.Core.Sync;
+
+public class LastSyncedManager : ILastSyncedManager
+{
+    private readonly ILastSyncedRepository  _lastSyncedRepository;
+    private readonly ICoreScopeProvider _coreScopeProvider;
+
+    public LastSyncedManager(ILastSyncedRepository lastSyncedRepository, ICoreScopeProvider coreScopeProvider)
+    {
+        _lastSyncedRepository = lastSyncedRepository;
+        _coreScopeProvider = coreScopeProvider;
+    }
+
+    public async Task<int?> GetLastSyncedInternalAsync()
+    {
+        using ICoreScope coreScope = _coreScopeProvider.CreateCoreScope();
+        int? internalId = await _lastSyncedRepository.GetInternal();
+        coreScope.Complete();
+
+        return internalId;
+    }
+
+    public async Task<int?> GetLastSyncedExternalAsync()
+    {
+        using ICoreScope coreScope = _coreScopeProvider.CreateCoreScope();
```

Comment:
We don't want to encode implementation details in variable names, so it's better if this is just called `scope` instead of `coreScope` :smile: